### PR TITLE
Decoupled W&B from training and added support for multiple loss functions (sum)

### PR DESCRIPTION
**Pull request by MrinalJain17**
- Modified W&B to be used as a callback. Moreover, changed the logging frequency of sample predictions. Fixed associated bugs.
- Added support to pass multiple loss functions, the sum of which is optimized (Eg: BCE + Dice).

### CODE REVIEW

File: capstone/data/datasets.py [hunk 1]
```diff
@@ -55,10 +56,13 @@ def __getitem__(self, index: int) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
...
+            masks = torch.from_numpy(np.expand_dims(masks[idx], axis=0))
+            mask_indicator = torch.from_numpy(
+                np.expand_dims(mask_indicator[idx], axis=0)
+            )
         else:
-            masks = np.stack(masks)
+            masks = torch.from_numpy(np.stack(masks))
+            mask_indicator = torch.from_numpy(mask_indicator)
 
         return image, masks, mask_indicator
```
Comment (MrinalJain17) [inline capstone/data/datasets.py:67]: The `masks` and `mask_indicator` were numpy arrays, and not tensors. Fixed that.

Although this should have no effect on the dataloaders.
